diff --git a/pcsx2-qt/QtHost.cpp b/pcsx2-qt/QtHost.cpp
index a177f39..8758bf9 100644
--- a/pcsx2-qt/QtHost.cpp
+++ b/pcsx2-qt/QtHost.cpp
@@ -18,6 +18,7 @@
 #include "pcsx2/CDVD/CDVD.h"
 #include "pcsx2/Counters.h"
 #include "pcsx2/DebugTools/Debug.h"
+#include "pcsx2/DebugTools/DebugServer.h"
 #include "pcsx2/GS.h"
 #include "pcsx2/GS/GS.h"
 #include "pcsx2/GSDumpReplayer.h"
@@ -384,6 +385,8 @@ void EmuThread::run()
 		return;
 	}
 
+	DebugServer::Start();
+
 	// Start background polling because the VM won't do it for us.
 	createBackgroundControllerPollTimer();
 	startBackgroundControllerPollTimer();
@@ -423,6 +426,7 @@ void EmuThread::run()
 	// Teardown in reverse order.
 	stopBackgroundControllerPollTimer();
 	destroyBackgroundControllerPollTimer();
+	DebugServer::Stop();
 	VMManager::Internal::CPUThreadShutdown();
 
 	// Move back to the UI thread, since we're no longer running.
diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index 4c8ce81..9a2b1cf 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -817,6 +817,7 @@ set(pcsx2DebugToolsSources
 	DebugTools/DisR5900asm.cpp
 	DebugTools/DisVU0Micro.cpp
 	DebugTools/DisVU1Micro.cpp
+	DebugTools/DebugServer.cpp
 	DebugTools/BiosDebugData.cpp)
 
 # DebugTools headers
@@ -835,6 +836,7 @@ set(pcsx2DebugToolsHeaders
 	DebugTools/DisASM.h
 	DebugTools/DisVUmicro.h
 	DebugTools/DisVUops.h
+	DebugTools/DebugServer.h
 	DebugTools/BiosDebugData.h)
 
 set(pcsx2HostSources
